src/engine/naive_engine.cc

```diff
@@ -55,20 +69,41 @@ class NaiveEngine final : public Engine {
     opr->opr_name = opr_name;
     return opr;
   }
+
   void DeleteOperator(OprHandle op) override {
     NaiveOpr *opr = op->Cast<NaiveOpr>();
     delete opr;
   }
+
   void Push(OprHandle op, Context exec_ctx, int priority = 0, bool profiling = false) override {
     NaiveOpr *opr = op->Cast<NaiveOpr>();
-    this->PushAsync(opr->fn,
-                    exec_ctx,
-                    opr->const_vars,
-                    opr->mutable_vars,
-                    opr->prop,
-                    priority,
-                    PROFILER_MESSAGE(opr->opr_name));
+    opr->profiling = profiling;
+    this->PushAsync([&](RunContext ctx, CallbackOnComplete on_complete) {
+#if MXNET_USE_PROFILER
```

Comment:
why was profiling in the interface. Shouldn't it be a member of engine? @ZihengJiang